test/Microsoft.AspNetCore.Mvc.TagHelpers.Test/CacheTagHelperTest.cs

```diff
@@ -648,6 +648,67 @@ public void UpdateCacheEntryOptions_SetsCachePreservationPriority()
             Assert.Equal(2, calls);
         }
 
+        [Fact]
+        public async Task ProcessAsync_ExceptionInProcessing_DoNotThrowInSubsequentRequests()
+        {
+            // Arrange
+            var id = "unique-id";
+            var childContent = "some-content";
+            var cache = new MemoryCache(new MemoryCacheOptions());
+
+            var tagHelperContext1 = GetTagHelperContext(id);
+            var tagHelperContext2 = GetTagHelperContext(id);
+
+            var tagHelperOutput1 = new TagHelperOutput(
```

Comment:
@sebastienros Nevertheless I have rewritten the test to use single helper/output and counter. Result is same - current dev branch causes NullReferenceException, test passes after fix